docs/topics/dokka-migration.md

```diff
@@ -77,157 +95,355 @@ After updating the Dokka Gradle plugin to v2, follow the migration steps applica
+```
 
-  New configuration:
+#### Output directory
 
-    ```kotlin
-    customAssets.from("example.png", "example2.png")
-    ```
+Use the `dokka {}` block to specify the output directory for the generated Dokka documentation.
 
-* **Output directory:** Use the `dokka {}` block to specify the output directory for generated Dokka documentation.
+Previous configuration:
+
+```kotlin
+tasks.dokkaHtml {
+    outputDirectory.set(layout.buildDirectory.dir("dokkaDir"))
+}
+```
 
-    Previous configuration:
+New configuration:
 
-    ```kotlin
-    tasks.dokkaHtml {
+```kotlin
+dokka {
+    dokkaPublications.html {
         outputDirectory.set(layout.buildDirectory.dir("dokkaDir"))
     }
-    ```
+}
+```
 
-    New configuration:
+#### Output directory for additional files
 
-    ```kotlin
-    dokka {
-        dokkaPublications.html {
-            outputDirectory.set(layout.buildDirectory.dir("dokkaDir"))
+The ability to set an output directory and include additional files, such as `README.md`, is still supported in
+DGP v2. However, you have to now configure it under `dokkaPublications.html` in the `dokka {}` block of the root project.
+
+Previous configuration:
+
+```kotlin
+tasks.dokkaHtmlMultiModule {
+    outputDirectory.set(rootDir.resolve("docs/api/0.x"))
+    includes.from(project.layout.projectDirectory.file("README.md"))
+}
+```
+
+New configuration:
+
+```kotlin
+dokka {
+    dokkaPublications.html {
+        outputDirectory.set(rootDir.resolve("docs/api/0.x"))
+        includes.from(project.layout.projectDirectory.file("README.md"))
+    }
+}
+```
+
+### Configure Dokka plugins
+
+Configuring built-in Dokka plugins with JSON has been deprecated in favor of a type-safe DSL. This change improves compatibility 
+with Gradle's incremental build system and improves task input tracking.
+
+Previous configuration:
+
+In DGP v1, Dokka plugins were configured manually using JSON. This approach caused issues with [registering task inputs](https://docs.gradle.org/current/userguide/incremental_build.html) 
+for Gradle up-to-date checks.
+
+Here is an example of the deprecated JSON-based configuration for the [Dokka Versioning plugin](https://kotl.in/dokka-versioning-plugin):
+
+```kotlin
+tasks.dokkaHtmlMultiModule {
+    pluginsMapConfiguration.set(
+        mapOf(
+            "org.jetbrains.dokka.versioning.VersioningPlugin" to """
+                { "version": "1.2", "olderVersionsDir": "$projectDir/dokka-docs" }
+                """.trimIndent()
+        )
+    )
+}
+```
+
+New configuration:
+
+DGP v2 replaces the JSON-based configuration with a type-safe DSL that is compatible with incremental builds.
+Use the `pluginsConfiguration{}` block to configure Dokka plugins in a type-safe way:
+
+```kotlin
+dokka {
+    pluginsConfiguration {
+        versioning {
+            version.set("1.2")
+            olderVersionsDir.set(projectDir.resolve("dokka-docs"))
         }
     }
-    ```
+}
+```
+
+### Configure custom Dokka plugins
```

Comment:
Why do we promote the feature in the migration guide? I expect the minimal required amount of material to move projects from DGP v1 to DGP v2.

Reply:
This is part of the feedback we got to update the migration guide. You can see in the [feedback issue](https://youtrack.jetbrains.com/issue/KT-72053/Dokka-DGP-v2-migration-guide-feedback) the comment addressed by @adam-enko as pointed [here.](https://github.com/Kotlin/dokka/issues/3869)